storage/client_raft_test.go

```diff
@@ -239,7 +241,9 @@ func TestFailedReplicaChange(t *testing.T) {
 	}()
 	storage.TestingCommandFilter = func(method string, args proto.Request, reply proto.Response) bool {
 		if method == proto.EndTransaction && args.(*proto.EndTransactionRequest).Commit == true {
-			reply.Header().SetGoError(util.Errorf("boom"))
+			oldErr := unsafe.Pointer(&reply.Header().Error)
+			newErr := proto.Error{Message: "boom"}
+			atomic.SwapPointer((*unsafe.Pointer)(oldErr), unsafe.Pointer(&newErr))
```

Comment:
Ah, you're correct. This is confusing. The first argument to `swapPointer` is a pointer to the pointer to set. But that needs to be a pointer to the `Error`, not a pointer to the `unsafe.Pointer`. Only minor tweak is that you can use `StorePointer` here instead of `SwapPointer` since you aren't using the return value.